Source/AwsCommonRuntimeKit/Utilities.swift

```diff
@@ -12,43 +12,48 @@ func zeroStruct<T>(_ ptr: UnsafeMutablePointer<T>) {
     memset(ptr, 0x00, MemoryLayout<T>.size)
 }
 
-extension Data {
-    @inlinable
-    var awsByteCursor: aws_byte_cursor {
-        return withUnsafeBytes { (rawPtr: UnsafeRawBufferPointer) -> aws_byte_cursor in
-            return aws_byte_cursor_from_array(rawPtr.baseAddress, self.count)
-        }
-    }
-}
-
 extension String {
-    @inlinable
-    var awsByteCursor: aws_byte_cursor {
-        return aws_byte_cursor_from_c_str(self.asCStr())
-    }
 
     public func base64EncodedMD5(allocator: Allocator = defaultAllocator, truncate: Int = 0) -> String? {
-        let input: UnsafePointer<aws_byte_cursor> = fromPointer(ptr: self.awsByteCursor)
-        let emptyBuffer: UInt8 = 0
-        let bufferPtr: UnsafeMutablePointer<UInt8> = fromPointer(ptr: emptyBuffer)
-        let buffer = aws_byte_buf(len: 0, buffer: bufferPtr, capacity: 16, allocator: allocator.rawValue)
-        let output: UnsafeMutablePointer<aws_byte_buf> = fromPointer(ptr: buffer)
-
-        guard AWS_OP_SUCCESS == aws_md5_compute(allocator.rawValue, input, output, truncate) else {
+        var buffer = aws_byte_buf()
+        if aws_byte_buf_init(&buffer, allocator.rawValue, 16) != AWS_OP_SUCCESS {
             return nil
         }
+        guard self.withByteCursorPointer({ strCursorPointer in
+           aws_md5_compute(allocator.rawValue, strCursorPointer, &buffer, truncate)
+        }) == AWS_OP_SUCCESS else {
+            return nil
```

Comment:
fail to clean up the aws_byte_buf if we fail here.
maybe do something like:
- use a `defer` block to ensure the byte_buf is cleaned up, regardless of where this function returns from
OR
- Create an internal helper `class ByteBuf` the calls clean up from its `deinit`

Reply:
or use the "no alloc" variant of aws_byte_buf:
```
var buffer_data = Array<UInt8>(repeating: 0, count: 16)
var buffer = aws_byte_buf_from_empty_array(....)
```

Reply:
Thanks, fixed.